snuba/settings/__init__.py

```diff
@@ -278,6 +255,45 @@
 COUNTER_WINDOW_SIZE = timedelta(minutes=10)
 
 
+# Slicing Configuration
+
+# Mapping of storage key to slice count
+# This is only for sliced storages
+SLICED_STORAGES: Mapping[str, int] = {}
+
+# Mapping storage key to a mapping of logical partition
+# to slice id
+LOGICAL_PARTITION_MAPPING: Mapping[str, Mapping[int, int]] = {}
+
+# The slice configs below are the "SLICED" versions to
+# the equivalent default settings above. For example,
+# "SLICED_KAFKA_TOPIC_MAP" is the "SLICED" version of
+# "KAFKA_TOPIC_MAP". These should be filled out
+# for any corresponding sliced storages defined above,
+# with the applicable number of slices in mind.
+
+# Storage set keys should be defined either in CLUSTERS
+# or SLICED_CLUSTERS. CLUSTERS will define clusters
+# which are not sliced, i.e. are associated with
+# only the default slice_id (0). CLUSTERS is defined in
+# the default way, without adding slice id in
+# the storage_sets field. We define sliced clusters,
+# i.e. clusters that reside on multiple slices
+# in SLICED_CLUSTERS. We define all associated
+# (storage set, slice id) pairs in SLICED_CLUSTERS
+# in the storage_sets field. Other fields are defined
+# in the same way as they are in CLUSTERS.
```

Comment:
I am not sure how this is going to work. 
For a sliced cluster, if the slice id is an integer, why is the key of the mapping at line 286 a string ?

Reply:
This typing follows from that of `CLUSTERS` above. The keys of the mappings are strings because these are the names of the fields. We kept it this way so that `SLICED_CLUSTERS` and `CLUSTERS` will have similar structure. There are examples in https://github.com/getsentry/snuba/blob/master/tests/clusters/test_cluster.py, but I can also try to revise this comment so that the definition/structure of `SLICED_CLUSTERS` is clearer.

Reply:
Thanks for the clarification, but I think the question remains. 
How, in this data structure, is a cluster associated to a slice number? Is the slice number a field in the mapping ? Or is the slice number implied by the order in the sequence? If it is the second how would we manage multiple sliced storages as different storages can have different number of slices ?

Reply:
I'm still thinking about your questions, but perhaps an example will make things clearer:

```
SLICED_CLUSTERS_CONFIG = [
    {
        "host": "host_slice",
        "port": 9000,
        "user": "default",
        "password": "",
        "database": "default",
        "http_port": 8123,
        "storage_set_slices": {("generic_metrics_distributions", 0)},
        "single_node": True,
    },
    {
        "host": "host_slice",
        "port": 9001,
        "user": "default",
        "password": "",
        "database": "slice_1_default",
        "http_port": 8124,
        "storage_set_slices": {("generic_metrics_distributions", 1)},
        "single_node": True,
    },
]
```
Here, the `storage_set_slices` field will contain tuples of the form (storage set key, slice id).

Reply:
Ah ok, so the schema changes as the storage_set list contains the slice for each storage set. Thanks. That answers my question.